launcher/Application.cpp

```diff
@@ -1367,7 +1367,7 @@ bool Application::openJsonEditor(const QString& filename)
     }
 }
 
-bool Application::launch(InstancePtr instance, bool online, bool demo, MinecraftTarget::Ptr targetToJoin, MinecraftAccountPtr accountToUse)
+bool Application::launch(InstancePtr instance, bool online, bool demo, bool askServerAddress, MinecraftTarget::Ptr targetToJoin, MinecraftAccountPtr accountToUse)
```

Comment:
This accumulation of parameters will be bothersome in the future.
Can't we pass some kind of structure or better for this PR: ask the server to join when you click the button and pass the targetToJoin?

Reply:
This would require putting more UI code in MinecraftInstance - I already put a fixme for having any

Reply:
Well is not like it doesn't have the UI code already there so you would not break anything.
But as above feel free to mark it resolved, but on this one, I do want the opinion of at least one other maintainer before I approve this.

Reply:
I'd rather see a struct here as well. 6 params is kinda icky

Reply:
yeah, a `LaunchOptions` struct with some functions to construct it with some smart defaults would be much preferable IMO

Reply:
It would just go from args to { args } though. Or would you prefer
```
LaunchOptions opts;
opts.online = true;
// etc.
```
But in any case I didn't want to do much refactoring in this PR

Reply:
I'd say just stick to a struct, as an enum is a lot less composable (which is probably something important to consider given we're still adding another launch mode after all this time)

Reply:
Sorry for the delay. It seems kind of hard to implement:
The launch function is in Application.h so would I put the launch options there? that's quite messy. I could also put it in LaunchController.h despite it not being used in the file at all. The most clean solution feels like moving the private fields of LaunchController into a struct like this
```c++
struct LaunchOptions {
    bool online = true;
    bool demo = false;
    bool askServerAddress = false;
    BaseProfilerFactory* profiler = nullptr;
    MinecraftAccountPtr accountToUse = nullptr;
    AuthSessionPtr session;
    MinecraftTarget::Ptr targetToJoin;
}

class LaunchController {
    // blah blah blah
   private:
    // ...
    LaunchOptions m_options;
}
```

but as I said the construction of the struct would look like { profiler, online, demo, askServerAddress, accountToUse, session, targetToJoin } anyway, unless you used a mutable local variable which I guess could be slightly more readable.

Reply:
This seems like a reasonable approach to me

Reply:
I will wait for the outcome of https://github.com/PrismLauncher/PrismLauncher/pull/2894 as it doesn't seem that necessary with only two bool options